tests/oidc/test_oidc_integration.py

```diff
@@ -0,0 +1,108 @@
+from django.conf import settings
+from django.http import HttpRequest
+from django.test import Client
+from django.urls import reverse
+
+import pytest
+from mozilla_django_oidc_db.models import UserInformationClaimsSources
+from responses import RequestsMock
+
+from digid_eherkenning.oidc.models import (
+    DigiDConfig,
+    DigiDMachtigenConfig,
+    EHerkenningBewindvoeringConfig,
+    EHerkenningConfig,
+)
+from digid_eherkenning.oidc.models.base import OpenIDConnectBaseConfig
+from digid_eherkenning.oidc.views import (
+    digid_init,
+    digid_machtigen_init,
+    eh_bewindvoering_init,
+    eh_init,
+)
+
+pytestmark = [
+    pytest.mark.skipif(not settings.OIDC_ENABLED, reason="OIDC integration disabled"),
+]
+
+
+def test_oidc_enabled(settings):
+    assert settings.OIDC_ENABLED is True
+
+
+@pytest.mark.parametrize(
+    "init_view,config_class",
+    (
+        (digid_init, DigiDConfig),
+        (eh_init, EHerkenningConfig),
+        (digid_machtigen_init, DigiDMachtigenConfig),
+        (eh_bewindvoering_init, EHerkenningBewindvoeringConfig),
+    ),
+)
+@pytest.mark.django_db
+def test_init_flow(
+    auth_request: HttpRequest,
+    mocked_responses: RequestsMock,
+    init_view,
+    config_class: type[OpenIDConnectBaseConfig],
+):
+    _name = config_class._meta.model_name
+    config = config_class(
+        enabled=True,
+        oidc_rp_client_id=f"client-{_name}",
+        oidc_rp_client_secret=f"secret-{_name}",
+        oidc_op_authorization_endpoint=f"http://oidc.example.com/start-auth/{_name}",
+    )
+    config.save()
+    mocked_responses.get(f"http://oidc.example.com/start-auth/{_name}", status=400)
```

Comment:
I'm not sure why the start-auth endpoint is mocked with status code 400 here, isn't this a test for the happy flow?

Reply:
ha! exactly because 400 is treated as available :)

we do the GET request without any parameters, and keycloak (?) then returns a HTTP 400 because the client ID/state/flow parameters are absent, but this doesn't mean the endpoint is not available. I saw the code explicitly allows for HTTP 400, so I added a test case for it.

Reply:
Aah that's right, I remember the IdP availability check now :+1: